File: count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int goodNodes = 0;
    public int goodNodes(TreeNode root) {
        dfs(root, root.val);

        return goodNodes;
    }

    private void dfs(TreeNode root, int max) {
        if (root != null) {
            if (max <= root.val) {
                goodNodes++;
            }
            dfs(root.left, Math.max(root.val, max));
            dfs(root.right, Math.max(root.val, max));
        }
    }
}

File: 1544-count-good-nodes-in-binary-tree/count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def helper(node, maximum):
            good = 0
            if node.val >= maximum:
                good += 1
            if node.left:
                good += helper(node.left, max(maximum, node.val))
            if node.right:
                good += helper(node.right, max(maximum, node.val))
            return good
        return helper(root, -float('inf'))